x/stakeibc/keeper/icqcallbacks_delegator_shares.go

```diff
@@ -26,108 +25,99 @@ import (
-		hostZone.ChainId, validator.Address, validator.DelegationAmt, validatorTokens))
+	// Calculate the number of tokens delegated (using the internal exchange rate)
+	delegatedTokens := queriedDelgation.Shares.Mul(validator.InternalExchangeRate.InternalTokensToSharesRate).TruncateInt()
+	k.Logger(ctx).Info(utils.LogICQCallbackWithHostZone(chainId, ICQCallbackID_Delegation,
+		"Previous Delegation: %v, Current Delegation: %v", validator.DelegationAmt, delegatedTokens))
 
 	// Confirm the validator has actually been slashed
-	if validatorTokens.Equal(validator.DelegationAmt) {
-		k.Logger(ctx).Info(fmt.Sprintf("DelegationCallback: Validator (%s) was not slashed", validator.Address))
+	if delegatedTokens.Equal(validator.DelegationAmt) {
+		k.Logger(ctx).Info(utils.LogICQCallbackWithHostZone(chainId, ICQCallbackID_Delegation, "Validator was not slashed"))
 		return nil
-	} else if validatorTokens.GT(validator.DelegationAmt) {
-		errMsg := fmt.Sprintf("DelegationCallback: Validator (%s) tokens returned from query is greater than the DelegationAmt", validator.Address)
-		k.Logger(ctx).Error(errMsg)
-		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, errMsg)
+	}
+	if delegatedTokens.GT(validator.DelegationAmt) {
+		return sdkerrors.Wrapf(sdkerrors.ErrInvalidRequest, "Validator (%s) tokens returned from query is greater than the DelegationAmt", validator.Address)
 	}
 
 	// TODO(TESTS-171) add some safety checks here (e.g. we could query the slashing module to confirm the decr in tokens was due to slash)
 	// update our records of the total stakedbal and of the validator's delegation amt
 	// NOTE:  we assume any decrease in delegation amt that's not tracked via records is a slash
 
 	// Get slash percentage
-	slashAmount := validator.DelegationAmt.Sub(validatorTokens)
-
-	weight, err := cast.ToInt64E(validator.Weight)
-	if err != nil {
-		errMsg := fmt.Sprintf("unable to convert validator weight to int64, err: %s", err.Error())
-		k.Logger(ctx).Error(errMsg)
-		return sdkerrors.Wrapf(types.ErrIntCast, errMsg)
-	}
-
+	slashAmount := validator.DelegationAmt.Sub(delegatedTokens)
 	slashPct := sdk.NewDecFromInt(slashAmount).Quo(sdk.NewDecFromInt(validator.DelegationAmt))
-	k.Logger(ctx).Info(fmt.Sprintf("ICQ'd Delegation Amount Mismatch, HostZone: %s, Validator: %s, Delegator: %s, Records Tokens: %v, Tokens from ICQ %v, Slash Amount: %v, Slash Pct: %v!",
-		hostZone.ChainId, validator.Address, queriedDelgation.DelegatorAddress, validator.DelegationAmt, validatorTokens, slashAmount, slashPct))
+	k.Logger(ctx).Info(utils.LogICQCallbackWithHostZone(chainId, ICQCallbackID_Delegation,
+		"Validator was slashed! Validator: %s, Delegator: %s, Delegation in State: %v, Delegation from ICQ %v, Slash Amount: %v, Slash Pct: %v",
+		validator.Address, queriedDelgation.DelegatorAddress, validator.DelegationAmt, delegatedTokens, slashAmount, slashPct))
 
 	// Abort if the slash was greater than 10%
 	tenPercent := sdk.NewDec(10).Quo(sdk.NewDec(100))
```

Comment:
While we're here, should we make this a param?

Reply:
Yeah! Do you remember all the steps for adding a param? We need to add it in our upgrade handler right?